completions/tmux

```diff
@@ -0,0 +1,297 @@
+        _comp_cmd_tmux__log "Unknown tmux subcommand: '$1'"
+        return
+    fi
+    local subcommand="${usage%% *}" # not $1, because it could be an alias
+    _comp_cmd_tmux__log "Attempting completion for 'tmux $subcommand'"
+
+    local -A options
+    local -a args
+    _comp_cmd_tmux__parse_usage "${usage#* }"
+
+    local option_type
+    local positional_start
+    _comp_cmd_tmux__options "${subcommand_args[@]}"
+
+    if [[ $option_type ]]; then
+        _comp_cmd_tmux__value "$subcommand" "$option_type"
+        return
+    elif ((positional_start < 0)) && [[ $cur == -* ]]; then
+        _comp_compgen -- -W '"${!options[@]}"'
+        return
+    elif ((positional_start < 0)); then
+        # $cur (one past the end of subcommand_args) is the first positional
+        # arg
+        positional_start=${#subcommand_args[@]}
+    fi
+
+    _comp_cmd_tmux__log \
+        "'tmux $subcommand' first positional arg: '${subcommand_args[positional_start]-}'"
+
+    local args_index="$positional_start"
+    local usage_args_index
+    local prev_arg_type=""
+    for ((\
+    usage_args_index = 0;  \
+    usage_args_index < ${#args[@]};  \
+    args_index++, usage_args_index++)); do
+        local arg_type="${args[usage_args_index]##+(\[)}"
+        arg_type="${arg_type%%+(\])}"
+        if [[ $arg_type == ... ]]; then
+            if ((usage_args_index == 0)); then
+                # Prevent an infinite loop.
+                _comp_cmd_tmux__log "'tmux $subcommand' first arg is '...'"
+                return
+            elif ((usage_args_index != ${#args[@]} - 1)); then
+                _comp_cmd_tmux__log \
+                    "'tmux $subcommand' usage has '...' before last arg"
+                return
+            fi
+            # Repeat from the beginning of args.
+            usage_args_index=-1
+            ((args_index--))
+        elif [[ $arg_type == arguments ]]; then
+            if [[ $prev_arg_type == command ]] &&
+                ((usage_args_index == ${#args[@]} - 1)); then
+                # The usage ends in `command arguments`, so recurse to the new
+                # subcommand.
+                _comp_cmd_tmux__subcommand \
+                    "${subcommand_args[@]:args_index-1}"
+                return
+            else
+                _comp_cmd_tmux__log \
+                    "'tmux $subcommand' has unsupported 'arguments' in usage"
+                return
+            fi
+        elif ((args_index == ${#subcommand_args[@]})); then
+            # The usage arg is 1 past the end of $subcommand_args, so complete
+            # it.
+            _comp_cmd_tmux__value "$subcommand" "$arg_type"
+            return
+        fi
+        prev_arg_type="$arg_type"
+    done
+
+    _comp_cmd_tmux__log "Too many args to 'tmux $subcommand'"
+}
+
+_comp_cmd_tmux()
+{
+    local cur prev words cword comp_args
+    _comp_initialize -- "$@" || return
+
+    local usage
+    usage="$(LC_ALL=C tmux -h 2>&1)"
```

Comment:
```suggestion
    usage="$(LC_ALL=C "$1" -h 2>&1)"
```

Reply:
Done, with `_comp_cmd_tmux__run` instead of `$1`.